src/main/java/com/_up/megastore/services/implementations/UserService.java

```diff
@@ -119,13 +121,13 @@ private User findUserToActivateOrThrowException(UUID userId,
 
   @Override
   public void sendEmailToRecoverPassword(String email) {
-    if (!userRepository.existsByEmail(email))
-      throw new RuntimeException("User not found");
-
     User user = findUserByEmailOrThrowException(email);
 
-    String recoverPasswordURL = frontendURL + "/auth/recover-password?userId=" + user.getUserId() +
-              "&token=" + user.getRecoverPasswordToken();
+    user.setRecoverPasswordToken(UUID.randomUUID());
+
+    userRepository.save(user);
+
+    String recoverPasswordURL = frontendURL + "/auth/recover-password?token=" + user.getRecoverPasswordToken();
```

Comment:
Pregunta, esto funciona? No tenes que hacer un .toString() o lo hace solo al querer concatenar dos strings?

```suggestion
    String recoverPasswordURL = frontendURL + "/auth/recover-password?token=" + user.getRecoverPasswordToken().toString();
```

Reply:
Lo hace solo prácticamente, porque lo probé y no dió ningún problema